### PR TITLE
Feature/bdi co  migration utility clone object mappings

### PR DESCRIPTION
**Pull request by psadie**
Add reference Object Mapping cloning to the Advanced Mapping Migration Utility.
Fixes a bug where the IsRequired field wasn't being carried over to migrated field mappings.

# Critical Changes

# Changes

# Issues Closed

# New Metadata
- Custom Metadata Type Custom Field
  - Data_Import_Object_Mapping__mdt.Legacy_Data_Import_Object_Name__c

Add default value to reference Object Mapping record Legacy_Data_Import_Object_Name__c fields (i.e. Data_Import_Object_Mapping.Account1.md.Legacy_Data_Import_Object_Name__c => 'Account1')
# Deleted Metadata

### CODE REVIEW

**Comment by psadie** [comment]
**lurch: attach W-034543

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001C7psQAC" target="_blank">W-034543</a>

File: src/classes/BDI_MigrationMappingHelper.cls [hunk 1]
```diff
@@ -130,10 +148,12 @@ public class BDI_MigrationMappingHelper {
                     Target_Object_Mapping__c,
                     Target_Object_Mapping__r.Object_API_Name__c,
                     Target_Object_Mapping__r.DeveloperName,
+                    Target_OBject_Mapping__r.Legacy_Data_Import_Object_Name__c,
                     Target_Object_Mapping__r.MasterLabel,
                     Is_Deleted__c
             FROM Data_Import_Field_Mapping__mdt
-            WHERE Data_Import_Field_Mapping_Set__r.DeveloperName = :developerName];
+            WHERE Data_Import_Field_Mapping_Set__r.DeveloperName = :developerName
+            AND Is_Deleted__c = false];
```
Comment (ptewson-sfdo) [inline src/classes/BDI_MigrationMappingHelper.cls]: Super minor comment, but can you indent this "AND" so that it's one tab further to the right?

Comment (psadie) [reply src/classes/BDI_MigrationMappingHelper.cls]: All set!

**Comment by psadie** [comment]
@ptewson-sfdo - Commits around protecting default field mapping, object mapping, field mapping set, and object mapping set are up

**Pull request merged**